### PR TITLE
obs-filters: Improve upward compressor & expander fix

### PR DESCRIPTION
**Pull request by pkviet**
### Description
This is a series of improvements for the upward compressor co-authored by @norihiro and @pkviet :
- gain above threshold was buggy and is corrected (pkv);
- gain is made to be continuous even at -60 dB (kamae & pkv);
- the hard knee is replaced by a soft knee (pkv);
Fixes:
#8052 

Furthermore, the hard setting of gain to 0 dB above threshold for the expander was removed.
Fixes:
#8030 

### Motivation and Context
Fix bugs for expander and upward compressor.
Enhance the upward compressor.

### How Has This Been Tested?
Tested to work by @norihiro : details [here](https://github.com/pkviet/obs-studio/pull/84) and @pkviet 

### Types of changes
- Bug fix (non-breaking change which fixes an issue) 
- Performance enhancement (non-breaking change which improves efficiency)

### Checklist:
- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Comment by RytoEX** [comment]
If this fixes a GitHub Issue, please use [closing keywords](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests) in the PR description.

**Comment by norihiro** [comment]
The knee is a new feature. I want to suggest to separate it at the end of the commit so that other fix commits can be cherry-picked to 29.0.1.

**Comment by norihiro** [comment]
I tested the knee feature. The kee looks working ok.

Input is a 440 Hz sine wave and sweeps the level using the `Gain` filter and changes the knee settings for 0, 10, and 20. Figure below shows the input level vs. the output level reading the level meter by eyes.
upwcomp-inout-level-knee

**Comment by RytoEX** [comment]
It seems like the first three commits here are bug fixes, while the fourth commit is a new feature. Is that correct? We may not make this in for 29.0.1, but we can reevaluate the bug fix commits for a potential 29.0.2, if needed.

**Comment by FelikZ** [comment]
can confirm that this PR fixes #8030 (for my usecase)

File: plugins/obs-filters/data/locale/en-US.ini [hunk 1]
```diff
@@ -115,6 +115,7 @@ Expander.None="None"
 Expander.Presets="Presets"
 Expander.Presets.Expander="Expander"
 Expander.Presets.Gate="Gate"
+expander.Knee.Width="Knee Width"
```
Comment (RytoEX) [inline plugins/obs-filters/data/locale/en-US.ini:118]: Nit: Expander

**Review by RytoEX — CHANGES_REQUESTED** [review]

File: plugins/obs-filters/data/locale/en-US.ini (hunk 1 above)
Comment (pkviet) [reply plugins/obs-filters/data/locale/en-US.ini]: fixed

**Pull request merged**